six/modules/c++/samples/test_create_sidd_from_mem.cpp

```diff
@@ -1773,17 +1774,498 @@ void initProcessorInformation(
+    err.components->radarSensor->clockFreqSF = 1111.1;
+    err.components->radarSensor->transmitFreqSF = 85;
+    err.components->radarSensor->rangeBiasDecorr.corrCoefZero = 123;
+    err.components->radarSensor->rangeBiasDecorr.decorrRate = .03;
+
+    err.components->tropoError.reset(new six::TropoError());
+    err.components->tropoError->tropoRangeVertical = .00289;
+    err.components->tropoError->tropoRangeSlant = 777;
+    err.components->tropoError->tropoRangeDecorr.corrCoefZero = 0;
+    err.components->tropoError->tropoRangeDecorr.decorrRate = 98.7;
+
+    err.components->ionoError.reset(new six::IonoError());
+    err.components->ionoError->ionoRangeVertical = 1.2;
+    err.components->ionoError->ionoRangeRateVertical = 77632;
+    err.components->ionoError->ionoRgRgRateCC = .072;
+    err.components->ionoError->ionoRangeVertDecorr.corrCoefZero = 48.16;
+    err.components->ionoError->ionoRangeVertDecorr.decorrRate = 113.964;
+
+    six::Parameter param;
+    param.setName("ErrorStatisticsParameterName");
+    param.setValue("Error StatisticsParameterValue");
+
+    err.additionalParameters.push_back(param);
+}
+
+void initRadiometric(six::Radiometric& radiometric)
+{
+    radiometric.noiseLevel.noiseType = "Noise type";
+    radiometric.noiseLevel.noisePoly = six::Poly2D(0, 0);
+    
+    radiometric.rcsSFPoly = six::Poly2D(0, 0);
+    radiometric.betaZeroSFPoly = six::Poly2D(1, 3);
+    radiometric.sigmaZeroSFPoly = six::Poly2D(0, 0);
+    radiometric.sigmaZeroSFIncidenceMap = six::AppliedType("IS_TRUE");
+    radiometric.gammaZeroSFPoly = six::Poly2D(0, 0);
+    radiometric.gammaZeroSFIncidenceMap = six::AppliedType("IS_FALSE");
+}
+
+void initAnnotations(six::sidd::Annotations& annotations)
+{
+    mem::ScopedCopyablePtr<six::sidd::Annotation> annotation(
+        new six::sidd::Annotation());
+    annotation->identifier = "Annotation identifier";
+
+    annotation->spatialReferenceSystem.reset(new six::sidd::SFAReferenceSystem());
+    annotation->spatialReferenceSystem->axisNames.push_back("Axis 1");
+    annotation->spatialReferenceSystem->coordinateSystem.reset(new six::sidd::SFAGeocentricCoordinateSystem());
+    six::sidd::SFAGeocentricCoordinateSystem* system = (six::sidd::SFAGeocentricCoordinateSystem*) annotation->spatialReferenceSystem->coordinateSystem.get();
+    system->csName = "CS Name";
+    system->datum.spheroid.name = "Spheroid";
+    system->datum.spheroid.semiMajorAxis = 69.03;
+    system->datum.spheroid.inverseFlattening = 71.473;
+    system->primeMeridian.name = "Prime Meridian";
+    system->primeMeridian.longitude = 11.11;
+    system->linearUnit = "Unit";
+
+    annotation->objects.resize(1);
+    annotation->objects[0].reset(new six::sidd::SFAPoint(2.4, 3.5, 4.55, 5.757));
+    annotations.push_back(annotation);
+}
+
+void populateData(six::sidd::DerivedData& siddData, const std::string&
+        lutType, const bool smallImage, const std::string& version)
```

Comment:
Input parameters passed by value don't need to be `const` (since the caller knows you can't modify them outside the scope of this function), so just send `smallImage` in as a `bool`